src/main/java/com/google/codeu/servlets/LocationServlet.java

```diff
@@ -0,0 +1,52 @@
+package com.google.codeu.servlets;
+
+import java.util.HashMap;
+import java.util.SortedMap;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.io.IOException;
+import com.google.codeu.data.Datastore;
+import com.google.codeu.data.Location;
+import com.google.gson.Gson;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/Location")
+public class LocationServlet extends HttpServlet {
+  private Datastore datastore;
+
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    List<LocationCount> loc = datastore.getLocationCount();
+    Collections.sort(loc, (x, y) -> {return -Integer.compare(x.count, y.count);});
+
+    Gson gson = new Gson();
+    String json = gson.toJson(loc);
+    response.getOutputStream().println(json);
+  }
+
+  public static class LocationCount {
```

Comment:
I'm thinking we can directly modify the Location class, add a count attribute, update the count in datastore whenever a new location is added in. So that we don't need to count them for every fetch request and wrap the result in this new class. But your code is also working good. I can do the change for Location class from my side after this PR is merged (coz I kind of desperately want this PR being merged and build upon your servlet haha 😃 )

Reply:
That's a good idea! Then I just merge it.